### PR TITLE
Contraband Drop 1 - Narcotics 1

### PR DESCRIPTION
**Pull request by KittenColony**
A new collection of PRs. _Contraband._
## About the PR
Adds several new items and sources for narcotics, including brand new ones with Unique* effects!

* New Cigarettes
Hack into a shady cigs dispenser, and activate the contraband catalog to enjoy 6 new packets and 1 new brand!
New Cigarette Brand ~ Garden Mirage. "Life's a Mirage, see through it."
New unmarked packets! Unknown, unnamed, unbelievably _good!_

* Hack vending machines for different narcotics.
Hack a Chefvend to get a hit of joy to keep you going, two bottles of Happiness.
Hack a sec-vend to get a leg-up over rebels and runners, new Dynamoxine autoinjectors give a burst of energy for ten seconds!

* New Chemicals
**Dynamoxine **
An attempt by Nanotrasen to create a military stimulant for extended engagements, shelved after Donk Co. came out with Hyperzine, the recipe has been made publicly available recently.
**Mindleaf**
A questionably legal plant with an intoxicating, toxic sap, which can be refined into a plethora of drugs, both potent and pleasant.

## Why / Balance
Characters might have a background with substance abuse, currently, the only real way to get these is through chemistry, spaceshrooms, maintpills or the occasional admin-run dealer events. This PR adds new sources for people to get their fixes, while also giving Security something to track down and interact with.

## Technical details
Primarily YAML code, does include several new reagents, reactions and FTL files, vending machine changes not name-spaced, are tagged with [# _CD]

## Media
image

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
add: Added a new cigarette brand laced with restricted chemicals.
add: Added a new collection of cigarette packets filled with potent narcotics.
add: Added Happiness pillbottles to the Chefvend
add: Added new Dynamoxine, an experimental stimulant, and Dynamoxine injectors to hacked Secvends
add: Added Mindleaf, a curious addition to Botany's supply, able to be turned into several potent drugs.

### CODE REVIEW

**Comment by KittenColony** [comment]
PR made to find issue with Dynamoxine using tests

File: Resources/Prototypes/_CD/Reagents/narcotics.yml
Comment (DrSmugleaf) [inline Resources/Prototypes/_CD/Reagents/narcotics.yml:31]: You are missing an indentation here, it is reading this as part of the conditions, not as part of the reagent threshold
```suggestion
            reagent: Epinephrine # DO NOT USE A FUCKING EPEN
            min: 1
```

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Welds asked me for a PR review in voice chat

**Comment by KittenColony** [comment]
motherfucking GOD, HOW?

**Comment by KittenColony** [comment]
I would like to apologise in advance to you, Aquif.

**Comment by KittenColony** [comment]
GOD FUCKING HELP ME I HATE CRAFTING CODE

**Comment by KittenColony** [comment]
fucking what

File: Resources/Prototypes/_CD/Recipes/Crafting/Graphs/narcotics.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Recipes/Crafting/Graphs/narcotics.yml:24]:
```suggestion
- type: constructionGraph
  id: WrappedBakePackage
  start: start
  graph:
  - node: start
    edges:
    - to: empty
      steps:
      - tag: EmptyPacket
        name: an empty packet
        doAfter: 1
  - node: empty
    entity: EmptyPacket
    edges:
    - to: bakeWrapped
      steps:
      - tool: Screwing
        doAfter: 1
      conditions:
      - !type:MinSolution
        solution: drink
        reagent:
          ReagentId: Bake
        quantity: 30
  - node: bakeWrapped
    entity: WrappedBakePackage

```

File: Resources/Prototypes/_CD/Recipes/Crafting/Graphs/package.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Recipes/Crafting/Graphs/package.yml:7]: `emptyPacket` instead

File: Resources/Prototypes/_CD/Recipes/Crafting/Graphs/package.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Recipes/Crafting/Graphs/package.yml:2]: `FoldSmugglerPackage`

File: Resources/Prototypes/_CD/Recipes/Crafting/Graphs/package.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Recipes/Crafting/Graphs/package.yml:12]: above

File: Resources/Prototypes/_CD/Entities/Objects/Misc/contraband.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Misc/contraband.yml:1]: `EmptyPacket` should have a way for you to add the solution that doesn't involve the addreagent command xP

File: Resources/Prototypes/_CD/Hydroponics/leaves.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Hydroponics/leaves.yml:18]: only one linebreak here

File: Resources/Prototypes/_CD/Reagents/narcotics.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Reagents/narcotics.yml:110]: doesn't need so many newlines

File: Resources/Prototypes/_CD/Reagents/narcotics.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Reagents/narcotics.yml:278]: above

File: Resources/Prototypes/_CD/Reagents/narcotics.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Reagents/narcotics.yml:573]: EOF newline

**Review by MilonPL — CHANGES_REQUESTED** [review]
you need to have a step at the `empty` node, i used the screwdriver but whatever, you got it from here I hope

**Comment by KittenColony** [comment]
1.0 is marked as complete and ready for review.

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml:52]: why does this need the suffix

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml:141]: use entitytables

File: Resources/Prototypes/_CD/Entities/Objects/Misc/contraband.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Misc/contraband.yml:47]: EOF newline

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml:72]: above

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml:136]: .

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml:68]: happsiness

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml:68]: pro

File: Resources/Prototypes/_CD/Recipes/Crafting/narcotics.yml
Comment (MilonPL) [inline Resources/Prototypes/_CD/Recipes/Crafting/narcotics.yml:13]: `folded` not `fold` or otherwise you will get `to create fold smuggling packet...`

**Review by MilonPL — CHANGES_REQUESTED** [review]
`type` > `abstract` > `parent` > `id` > `name` > `suffix` > `description` > `components`
yaml conventions everywhere jfc

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml
Comment (KittenColony) [reply Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/cigarette.yml:68]: intentional
happiness + Psicodine

File: Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml
Comment (KittenColony) [reply Resources/Prototypes/_CD/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml:52]: clinging to uniformity

**Comment by dffdff2423** [comment]
It looks like you marked the above review from Milon as resolved but I don't see the actual changes. Did you forget to push them?

**Comment by KittenColony** [comment]
I'm not going to rewrite every YAML entity for the sake of convention when the vast majority of the existing code doesn't follow it

it was a pain in the ass to even get the random pack vending to work at all, and what's already there was copied from how the random jumpsuits are dispensed

HapPsiness is intentional

**Comment by LankLTE** [comment]
Test fail is related I believe

**Comment by KittenColony** [comment]
image

**Review by dffdff2423 — COMMENTED** [review]
We should be moving in the direction of following the conventions, even if old code does not. Having a random order for entity members for each entity damages readability because it requires more effort to see if an entity has a given member. `entityTable`s are preferred because they allow more configuration on how random values are chosen.

We are currently discussing the mechanics of the reagents privately. I will get back to you shortly on those.

**Comment by dffdff2423** [comment]
I reran the test just in-case (although it looks like Lank did already).

**Comment by KittenColony** [comment]
as an aside with the reagents, all of them barring the sap and tea should be considered restricted chemicals / as non-insignificant contraband by the crew / security

**Comment by dffdff2423** [comment]
Currently blocked awaiting private maintainer discussion. I will update you as soon as possible.

**Comment by KittenColony** [comment]
Are you fucking kidding me

**Review by Nairodian — APPROVED** [review]
Okay I think this is alright. Next time you make a PR that adds/changes a lot I ask that you be more descriptive in your description about what the items actually functionally do instead of just a vague overview.

**Comment by dffdff2423** [comment]
Next time you make a pull request please break it up into multiple smaller PRs so it does not take a month to review. I don't care if you spam the repo with 20 PRs. It is far easier to review because I don't need to set aside an hour and I can consider each thing in isolation rather than all at once.

File: Resources/Prototypes/_CD/Entities/Objects/Misc/contraband.yml
Comment (LankLTE) [reply Resources/Prototypes/_CD/Entities/Objects/Misc/contraband.yml:41]: Just remember to lower this if it ends up causing problems.

**Comment by LankLTE** [comment]
Will merge this after the upstream merge.

**Pull request merged**
